collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java

```diff
@@ -132,6 +134,9 @@ public void granted() {
 
                 @Override
                 public void denied() {
+                    if (activity != null && (activity instanceof GoogleSheetsUploaderActivity || activity instanceof GoogleDriveActivity)) {
```

Comment:
Ohh sorry I didn't catch it before but in fact, you don't need that first null check here:
https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

your IDE should highlight that check and let you know it's not needed.